storage-node/src/cache/lru.rs

```diff
@@ -1,38 +1,178 @@
+impl<K, V> LruCache<K, V>
+where
+    K: Hash + Eq + Clone + Debug,
+{
+    pub fn new(max_capacity: usize) -> LruCache<K, V> {
+        LruCache {
+            cache_map: LinkedHashMap::new(),
+            max_capacity,
+            curr_capacity: 0,
+        }
+    }
+
+    fn lru_get(&mut self, key: &K) -> Option<&V> {
+        match self.cache_map.raw_entry_mut().from_key(key) {
+            linked_hash_map::RawEntryMut::Occupied(mut entry) => {
+                entry.to_back();
+                Some(entry.into_mut())
+            }
+            linked_hash_map::RawEntryMut::Vacant(_) => None,
+        }
+    }
+
+    fn lru_peek(&self, key: &K) -> Option<&V> {
+        self.cache_map.get(key)
+    }
+}
+
+impl ParpulseCache for LruCache<ParpulseCacheKey, ParpulseCacheValue> {
+    fn get(&mut self, key: &ParpulseCacheKey) -> Option<&ParpulseCacheValue> {
+        self.lru_get(key)
+    }
+
+    fn put(&mut self, key: ParpulseCacheKey, value: ParpulseCacheValue) {
+        // If the file size is greater than the max capacity, return
+        if value.1 > self.max_capacity {
+            println!(
+                "Warning: The file size of key {:?} is greater than the max capacity",
+                key
+            );
+            println!(
+                "File size: {:?}, Max capacity: {:?}",
+                value.1, self.max_capacity
+            );
+            return;
+        }
+        // If the key already exists, update the file size
+        if let Some((_, file_size)) = self.cache_map.get(&key) {
+            self.curr_capacity -= file_size;
+        }
+        self.curr_capacity += value.1;
+        self.cache_map.insert(key.clone(), value);
+        while self.curr_capacity > self.max_capacity {
+            if let Some((key, (_, file_size))) = self.cache_map.pop_front() {
+                println!("-------- Evicting Key: {:?} --------", key);
+                self.curr_capacity -= file_size;
+            }
+        }
+    }
+
+    fn peek(&self, key: &ParpulseCacheKey) -> Option<&ParpulseCacheValue> {
+        self.lru_peek(key)
     }
 
-    pub fn get(&mut self, key: &K) -> Option<V> {
-        todo!()
+    fn len(&self) -> usize {
+        self.cache_map.len()
     }
 
-    pub fn put(&mut self, key: K, value: V) -> bool {
-        todo!()
+    fn is_empty(&self) -> bool {
+        self.cache_map.is_empty()
     }
 
-    pub fn pop(&mut self) -> Option<V> {
-        todo!()
+    fn size(&self) -> usize {
+        self.curr_capacity
+    }
+
+    fn get_max_capacity(&self) -> usize {
+        self.max_capacity
+    }
+
+    fn set_max_capacity(&mut self, capacity: usize) {
+        self.max_capacity = capacity;
+    }
+
+    fn clear(&mut self) {
+        self.cache_map.clear();
+        self.curr_capacity = 0;
     }
 }
 
-impl ParpulseCache for LruCache<ParpulseCacheKey, ParpulseCacheValue> {
-    fn get_value(&mut self, key: &ParpulseCacheKey) -> Option<ParpulseCacheValue> {
-        self.get(key)
+mod tests {
```

Comment:
```suggestion
#[cfg(test)]
mod tests {
```
This allows the compiler to compile the mod only when testing.